packages/api/src/platforms/vtex/resolvers/offer.ts

```diff
@@ -17,98 +17,93 @@ type SearchProduct = ArrayElementType<
 >
 type Root = SearchProduct | OrderFormProduct
 
-const isSearchItem = (item: any): item is Item => 'sellers' in item
-const isOrderFormItem = (item: any): item is OrderFormProduct =>
-  'skuName' in item
+const isSearchItem = (item: Root): item is SearchProduct =>
+  'Price' in item && 'seller' in item && 'product' in item
 
-const getAvailability = (available: boolean) =>
-  available ? 'https://schema.org/InStock' : 'https://schema.org/OutOfStock'
+const isOrderFormItem = (item: Root): item is OrderFormProduct =>
+  'skuName' in item
 
 export const StoreOffer: Record<string, Resolver<Root>> = {
   priceCurrency: () => '',
-  priceValidUntil: (item) => {
-    if (isSearchItem(item)) {
-      return getFirstSeller(item.sellers)?.commertialOffer.PriceValidUntil ?? ''
+  priceValidUntil: (root) => {
```

Comment:
Wdyt about adding a better name for the first resolver parameter instead of root? For example: `enhancedCommercialOffer` or `commercialOffer`

Reply:
we use like this on all other resolvers. It's a common practice name the first arg of a resolver as root/parent/obj. Take a look at https://www.apollographql.com/docs/apollo-server/data/resolvers/ and https://graphql.org/learn/execution/#root-fields-resolvers